lib/java/com/google/android/material/lists/MaterialViewHolder.java

```diff
@@ -0,0 +1,14 @@
+package com.google.android.material.lists;
+
+import android.view.View;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+/** A standard view holder that all specific Material view holders extend */
+public abstract class MaterialViewHolder extends RecyclerView.ViewHolder {
```

Comment:
I do agree with your point about ```e.printstacktrace()``` but here is the situation (excluding the NoSuchMethodException):
* These three errors are not meant to be thrown and there are no reasonable circumstances where they can be thrown. If you are able to call the constructor -> no error. If you cannot get the constructor -> NoSuchMethodException.
* These errors should not "bubble up" because in these cases the developer did not do anything wrong. 

What would be a better way to handle these errors @BeksOmega? Maybe we should bubble them up? How can we handle the errors more elegantly?

Reply:
Regarding the ```MaterialViewHolder``` right now it is just for organization. In the future it will include real functionality when we have to stuff like reordering and selection.

Good question, it has previously been unclear.

Reply:
So I found the [AOSP guide for error handling](https://source.android.com/setup/contribute/code-style) (which I assume applies here?). From my reading, I believe if you don't expect the errors to be thrown the best thing to do is let them bubble up. But you're more familiar with this area so I think it's better if you give it a look.

Reply:
Nice research, didn't even know that existed! After looking at the method definition for getConstructor() and newInstance() I learned that the three other exceptions are called if the class is configured incorrectly.

Therefore I am going to Log.e a description for each error when it happens. It is the equivalent of this option:

<img width="863" alt="Screen Shot 2019-12-18 at 9 20 08 PM" src="https://user-images.githubusercontent.com/24931732/71146995-41484b00-21dc-11ea-8feb-9aa87afcd5f2.png">

- [X] Errors do occur (When class is configured incorrectly)
- [X] It should crash (The app has not setup their custom view holder correctly)

To clarify, these crashes only happen if the app defines a custom view holder. If the app uses our view holders it will be fine 😎

Reply:
In the final implementation, I used RuntimeException instead of Log.e since it seems like the more recommended way to error crash.